### PR TITLE
fix: [CDS-51795]: A/O Service Environment UI Issues

### PR DESCRIPTION
**Pull request by vikrant-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by vikrant-harness** [comment]
retrigger titlecheck

**Review by kapilkumawat86 — DISMISSED** [review]

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureModal.tsx [hunk 1]
```diff
@@ -261,6 +261,9 @@ function BootstrapDeployInfraDefinition({
   })
   const formikRef = useRef<FormikProps<Partial<InfrastructureDefinitionConfig>>>()
   const { stage } = getStageFromPipeline<DeploymentStageElementConfig>(selectedStageId || '')
+  const scope = projectIdentifier
+    ? getScopeFromValue(environmentIdentifier)
+    : getScopeFromDTO({ accountId, orgIdentifier, projectIdentifier })
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureModal.tsx:266]: Would this not give the scope for project as well?

Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureModal.tsx:266]: What is this doing different? Can you add a comment as well please

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx [hunk 2]
```diff
@@ -299,9 +302,13 @@ export default function DeployEnvironment({
     prependEnvironmentToEnvironmentList(newEnvironmentInfo)
     closeAddNewModal()
 
+    const scope = projectIdentifier ? 'project' : orgIdentifier ? 'org' : 'account'
+    const scopedEnvRef =
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx:306]: Can we move this to a util?

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx [hunk 3]
```diff
@@ -294,20 +294,23 @@ export default function DeployServiceEntityWidget({
 
     // add the new service to selection
     /* istanbul ignore else */
+
+    const scope = projectIdentifier ? 'project' : orgIdentifier ? 'org' : 'account'
+    const scopedServiceRef = scope !== 'project' ? `${scope}.${newServiceInfo.identifier}` : newServiceInfo.identifier
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx:299]: move to util

Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx]: Refactored this

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx]: Refactored

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureModal.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureModal.tsx]: Added a new prop for the same from the parent

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**